src/hub.js

```diff
@@ -1115,46 +1113,86 @@ document.addEventListener("DOMContentLoaded", async () => {
-          authChannel.setSocket(socket);
-          linkChannel.setSocket(socket);
-
-          // Disconnect old socket after a delay to ensure this user is always registered in presence.
-          setTimeout(() => {
-            console.log("Reconnection complete. Disconnecting old reticulum socket.");
-            oldSocket.teardown();
-          }, 10000);
+  const migrateToNewReticulumServer = async ({ ret_version, ret_pool }) => {
+    console.log(`[reconnect] Reticulum deploy detected v${ret_version} on ${ret_pool}.`);
+
+    const maxAttempts = 4;
+    let didReconnect = false;
+    let attempt = 0;
+    while (!didReconnect && attempt < maxAttempts) {
+      try {
+        // Add randomness to avoid flooding reticulum.
+        const delayMS = attempt * 3000 + Math.random() * 15000;
+        console.log(
+          `[reconnect] Attempting reconnect in ${Math.ceil(delayMS / 1000)} seconds.${
+            attempt ? ` (Attempt ${attempt + 1} of ${maxAttempts})` : ""
+          }`
+        );
+        await sleep(delayMS);
+        // Reconnect until reticulum meta matches expected version and pool.
+        didReconnect = await reconnect({ ret_version, ret_pool });
+      } catch {
+        didReconnect = false;
+      }
+
+      attempt = attempt + 1;
+    }
+
+    if (!didReconnect) {
+      console.error("[reconnect] Failed to reconnect to reticulum.");
+      return;
+    }
+
+    console.log("[reconnect] Reconnect in progress. Updated reticulum meta.");
+    const oldSocket = retPhxChannel.socket;
+    const socket = await connectToReticulum(isDebug, oldSocket.params());
+    retPhxChannel = await migrateChannelToSocket(retPhxChannel, socket);
+    await hubChannel.migrateToSocket(socket, createHubChannelParams());
+    authChannel.setSocket(socket);
+    linkChannel.setSocket(socket);
+
+    // Disconnect old socket after a delay to ensure this user is always registered in presence.
+    await sleep(10000);
+    oldSocket.teardown();
+    console.log("[reconnect] Reconnection successful.");
+  };
+
+  const onRetDeploy = (function() {
+    const notificationQueue = [];
+
+    const handleNextMessage = (function() {
+      let isLocked = false;
+      return async function handleNextMessage() {
+        if (isLocked || !notificationQueue.length) return;
+
+        isLocked = true;
+        const [notification] = notificationQueue.splice(0, 1);
+        try {
+          await migrateToNewReticulumServer(notification);
+        } catch {
+          console.error("Failed to migrate to new reticulum server after deploy.", notification);
+        } finally {
+          isLocked = false;
+          handleNextMessage();
         }
       };
+    })();
 
-      retDeployReconnectInterval = setInterval(tryReconnect, 5000);
-      tryReconnect();
-    }, Math.floor(Math.random() * retReconnectMaxDelayMs));
-  };
+    return function onRetDeploy(deployNotification) {
+      notificationQueue.push(deployNotification);
```

Comment:
Hmm, queuing these feels a bit odd. If we happen to do multiple deploys back to back the previous ones are now invalid and will never resolve, so we are guaranteed the reconnect attempt will fail, so it might make sense to short circuit them and save us some load.